app/streamSelector/browser.js

```diff
@@ -0,0 +1,87 @@
+const { desktopCapturer, ipcRenderer } = require("electron");
+window.addEventListener("DOMContentLoaded", () => {
+  ipcRenderer.once('get-screensizes-response', (event, screens) => {
+    let windowsIndex = 0;
+    const sscontainer = document.getElementById('screen-size');
+    screens.forEach((s, i) => {
+      const opt = document.createElement('option');
+      opt.appendChild(document.createTextNode(s.name));
+      opt.value = i;
+      sscontainer.appendChild(opt);
+    });
+    let defaultSelection = screens.findIndex(s => {
+      return s.default;
+    });
+
+    defaultSelection = defaultSelection > -1 ? defaultSelection : screens.length - 1;
+    sscontainer.selectedIndex = defaultSelection;
+
+    const _tabClick = (e) => {
+      document.querySelectorAll('button').forEach(b => {
+        b.classList.toggle('btn-primary');
+        b.classList.toggle('btn-secondary');
+      });
+      document.querySelector('.container-fluid').setAttribute('data-view', e.target.getAttribute('data-view'));
+    };
+
+    const _closePreviews = () => {
+      const vidElements = document.getElementsByTagName('video');
+      for (const vidElement of vidElements) {
+        vidElement.pause();
+        vidElement.srcObject.getVideoTracks()[0].stop();
+      };
+    };
+
+    document.querySelector('#btn-screens').addEventListener("click", _tabClick);
+    document.querySelector('#btn-windows').addEventListener("click", _tabClick);
+    document.querySelector('#btn-close').addEventListener("click", () => {
+      _closePreviews();
+      ipcRenderer.send("close-view");
+    });
+    desktopCapturer.getSources({ types: ['window', 'screen'] }).then(async sources => {
+      const rowElement = document.querySelector('.container-fluid .row');
```

Comment:
all of this can be (and probably should be) in a separate function, just to make it easier to read/maintain/etc.